src/modules/market/actions/load-full-market.js

```diff
@@ -7,12 +7,16 @@ import { addMarketLoading, removeMarketLoading } from 'modules/market/actions/up
 export const loadFullMarket = marketID => (dispatch, getState) => {
   dispatch(addMarketLoading(marketID));
 
+  // TODO -- improve conditional statement, leaving more verbose loading for now
```

Comment:
I was getting periodic race conditions between when the callback was being called and when the state was actually updated.